header.go

```diff
@@ -57,41 +84,17 @@ func mergeHeaders(into, from http.Header) {
 	}
 }
 
-// mergeMetdataHeaders merges the metadata headers from the "from" header into
-// the "into" header. It skips over non metadata headers that should not be
-// propagated from the server to the client.
-func mergeMetadataHeaders(into, from http.Header) {
+// mergeNonProtocolHeaders merges headers excluding protocol headers defined in
+// protocolHeaders.
+func mergeNonProtocolHeaders(into, from http.Header) {
```

Comment:
How about if we also rename the other one `mergeAllHeaders` and add the doc comment back to it, but clarifying that it adds _all_ header values, including protocol headers?

Reply:
I think `mergeHeaders` implies all.